src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -149,12 +166,50 @@ static grpc_status_code derive_nonce(uint16_t ciphersuite, uint8_t* secret,
                        secret_size, output_size, output, error_details);
 }
 
-static grpc_status_code assign_crypter(bool in, uint8_t* traffic_secret,
-                                       size_t traffic_secret_size,
-                                       size_t tag_size, uint64_t sequence,
-                                       s2a_crypter** crypter,
-                                       char** error_details) {
+/** This method populates |aead_crypter| with a gsec_aead_crypter instance and
+ *  returns GRPC_STATUS_OK on success; if |aead_crypter| points to a non-null
+ *  gsec_aead_crypter, then this method first destroys that crypter. The caller
+ *  must not pass in nullptr for |aead_crypter|. On failure,
+ *  the method returns an error code and populates |error_details|, which must
+ *  be freed using gpr_free. **/
+static grpc_status_code assign_aead_crypter(uint16_t ciphersuite, uint8_t* key,
+                                            size_t key_size, size_t nonce_size,
+                                            size_t tag_size,
+                                            gsec_aead_crypter** aead_crypter,
+                                            char** error_details) {
+  GPR_ASSERT(aead_crypter != nullptr);
   GPR_ASSERT(error_details != nullptr);
+  if (*aead_crypter != nullptr) {
+    gsec_aead_crypter_destroy(*aead_crypter);
+  }
+  grpc_status_code aead_crypter_status;
+  switch (ciphersuite) {
+    case kTlsAes128GcmSha256:
+    case kTlsAes256GcmSha384:
+      aead_crypter_status = gsec_aes_gcm_aead_crypter_create(
+          key, key_size, nonce_size, tag_size,
+          /* rekey=*/false, aead_crypter, error_details);
```

Comment:
Please remove the extra space in the comment.

Reply:
Done. And I did the same for other instances in this file. I will later do a pass of the entire S2A client to update to the same convention as the S2A.